src/main/java/de/tum/in/www1/artemis/web/rest/UserJWTController.java

```diff
@@ -115,6 +133,43 @@ public ResponseEntity<JWTToken> authorizeSAML2(@RequestBody final String body) {
         return new ResponseEntity<>(new JWTToken(jwt), httpHeaders, HttpStatus.OK);
     }
 
+    /**
+     * Generates a personal access token (JWT token) for the current user.
+     *
+     * @param lifetimeMilliseconds the lifetime for the JWT token
+     * @return a JWT Token
+     */
+    @PostMapping("/personal-access-token")
+    @FeatureToggle(Feature.PersonalAccessTokens)
+    public ResponseEntity<JWTToken> getPersonalAccessToken(@RequestBody Long lifetimeMilliseconds) {
+        User user = userRepository.getUser();
+
+        if (!user.getActivated()) {
+            throw new UserNotActivatedException("User was disabled!");
+        }
+
+        long difference = this.personalAccessTokenMaxLifetimeMilliseconds - lifetimeMilliseconds;
+        if (lifetimeMilliseconds > this.personalAccessTokenMaxLifetimeMilliseconds) {
+            return ResponseEntity.badRequest().headers(HeaderUtil.createFailureAlert(applicationName, true, ENTITY_NAME, "invalidPATLifetime",
+                    "Requested token lifetime exceeds the maximum lifetime for personal access tokens by " + difference)).build();
+        }
+
+        // Automatically returns 401 if not fully authorized
+        Authentication authentication = SecurityContextHolder.getContext().getAuthentication();
+        final String jwt = tokenProvider.createTokenWithCustomDuration(authentication, lifetimeMilliseconds);
+        return new ResponseEntity<>(new JWTToken(jwt), HttpStatus.OK);
+    }
+
+    /**
+     * Returns the maximum lifetime for personal access tokens.
+     * @return the maximum lifetime for personal access tokens
+     */
+    @GetMapping("/personal-access-token")
```

Comment:
The naming of the endpoint might be a bit confusing since the GET request does not actually return a token but only the maximum allowed lifetime.